### PR TITLE
Add OS BBE and refactor Log, UUID, and Randoms BBEs

### PR DESCRIPTION
**Pull request by MadhukaHarith92**

### CODE REVIEW

File: examples/log-api/log_api.bal [hunk 1]
```diff
@@ -4,26 +4,10 @@ public function main() {
...
-    // Each module can also be assigned its own log level. To assign a
-    // log level to a module, provide the following entry in the Config.toml file:
-    //
-    // ```
-    // [[ballerina.log.modules]]
-    // name = "[ORG_NAME]/[MODULE_NAME]"
-    // level = "[LOG_LEVEL]"
-    // ```
 
     // Users can pass any number of key/value pairs which needs to be displayed in the log message.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:12]:
```suggestion
    // You can pass any number of key/value pairs, which need to be displayed in the log message.
```

File: examples/log-api/log_api.bal [hunk 2]
```diff
@@ -4,26 +4,10 @@ public function main() {
...
     // These can be of `anydata` type including int, string and boolean.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:13]:
```suggestion
    // These can be of the `anydata` type including int, string, and boolean.
```

File: examples/log-api/log_api.bal [hunk 3]
```diff
@@ -36,13 +20,33 @@ public function main() {
         id = isolated function() returns int { return 845315;}, name = "foo");
 
     // Users can also pass a key/value pair where the value is an error stack trace.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:22]:
```suggestion
    // You can also pass a key/value pair in which the value is an error stack trace.
```

File: examples/log-api/log_api.bal [hunk 4]
```diff
@@ -36,13 +20,33 @@ public function main() {
...
+    log:printError("error log", stackTrace = err.stackTrace().callStack,
+     username = "Alex92", id = 845315);
 
     // Optionally, an error can be passed to the functions.
     error e = error("something went wrong!");
     log:printError("error log with cause", 'error = e, id = 845315,
         name = "foo");
 
+    // The log level can be configured via a Ballerina configuration file.
+    // To set the global log level, place the entry given below in the Config.toml file:
```
Comment (praneesha) [inline examples/log-api/log_api.bal:33]:
```suggestion
    // To set the global log level, place the entry given below in the `Config.toml` file:
```
Need to change to display the TOML file separately.

File: examples/log-api/log_api.bal [hunk 5]
```diff
@@ -36,13 +20,33 @@ public function main() {
...
+    //
+    // ```
+    // [ballerina.log]
+    // level = "[LOG_LEVEL]"
+    // ```
+
+    // Each module can also be assigned its own log level. To assign a
+    // log level to a module, provide the following entry in the Config.toml file:
```
Comment (praneesha) [inline examples/log-api/log_api.bal:41]:
```suggestion
    // log level to a module, provide the following entry in the `Config.toml` file:
```

Need to change to display the TOML file separately.

File: examples/log-api/log_api.bal [hunk 6]
```diff
@@ -36,13 +20,33 @@ public function main() {
...
+    //
+    // ```
+    // [[ballerina.log.modules]]
+    // name = "[ORG_NAME]/[MODULE_NAME]"
+    // level = "[LOG_LEVEL]"
+    // ```
+
+    // By default, log messages are logged to the console in LogFmt format.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:49]:
```suggestion
    // By default, log messages are logged to the console in the `LogFmt` format.
```

File: examples/log-api/log_api.bal [hunk 7]
```diff
@@ -36,13 +20,33 @@ public function main() {
...
     // To set the output format to JSON, place the entry given below in the `Config.toml` file.
```
Comment (praneesha) [inline examples/log-api/log_api.bal:50]:
```suggestion
    // To set the output format to JSON, place the entry given below in the `Config.toml` file.
```
Need to change to display the TOML file separately.

File: examples/os/os.bal [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import ballerina/io;
+import ballerina/os;
+
+public function main() {
+    // Returns the environment variable value associated with "HTTP_PORT".
```
Comment (praneesha) [inline examples/os/os.bal:5]:
```suggestion
    // Returns the environment variable value associated with the `HTTP_PORT`.
```

File: examples/os/os.bal [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    string port = os:getEnv("HTTP_PORT");
+    io:println("HTTP_PORT: ", port);
+
+    // Returns the current user's name.
```
Comment (praneesha) [inline examples/os/os.bal:9]:
```suggestion
    // Returns the username of the current user.
```

File: examples/os/os.description [hunk 10]
```diff
@@ -0,0 +1,3 @@
+// The `os` library provides functions to retrieve information about the os and the current users of the os.<br/><br/>
```
Comment (praneesha) [inline examples/os/os.description:1]:
```suggestion
// The `os` library provides functions to retrieve information about the OS and the current users of the OS.<br/><br/>
```

File: examples/random/random.bal [hunk 11]
```diff
@@ -1,12 +1,12 @@
 import ballerina/io;
 import ballerina/random;
 
-public function main() {
+public function main() returns error? {
     // Generate a random decimal number between 0.0 and 1.0.
```
Comment (praneesha) [inline examples/random/random.bal:5]:
```suggestion
    // Generates a random decimal number between 0.0 and 1.0.
```

File: examples/random/random.bal [hunk 12]
```diff
@@ -1,12 +1,12 @@
...
     float randomDecimal = random:createDecimal();
     io:println("Random decimal number: ", randomDecimal);
 
     // Generate a random number between the given start(inclusive) and end(exclusive) values.
```
Comment (praneesha) [inline examples/random/random.bal:9]:
```suggestion
    // Generates a random number between the given start(inclusive) and end(exclusive) values.
```

File: examples/uuid/uuid.bal [hunk 13]
```diff
@@ -1,68 +1,66 @@
...
     io:println("Nil UUID as a record: ", nilUuidRecord);
 
-    // Test a string to see if it is a valid UUID.
+    // Tests a string to see if it is a valid UUID.
     boolean valid = uuid:validate("4397465e-35f9-11eb-adc1-0242ac120002");
     io:println("UUID validated: ", valid.toString());
 
-    // Detect RFC version of a UUID.
-    uuid:Version|uuid:Error v = uuid:getVersion(
+    // Detects RFC version of a UUID.
```
Comment (praneesha) [inline examples/uuid/uuid.bal:53]:
```suggestion
    // Detects the RFC version of a UUID.
```

File: examples/log-api/log_api.bal (hunk 3 above)
Comment (MadhukaHarith92) [reply examples/log-api/log_api.bal:22]: Shouldn't we avoid "You", "We", etc?

File: examples/log-api/log_api.out [hunk 14]
```diff
@@ -2,21 +2,21 @@
...
+{"time":"2021-05-14T09:37:05.248+05:30", "level":"ERROR", "module":"", "message":"error log"}
+{"time":"2021-05-14T09:37:05.263+05:30", "level":"INFO", "module":"", "message":"info log"}
+{"time":"2021-05-14T09:37:05.264+05:30", "level":"WARN", "module":"", "message":"warn log"}
+{"time":"2021-05-14T09:37:05.265+05:30", "level":"INFO", "module":"", "message":"info log", "name":"foo", "id":845315, "successful":true}
+{"time":"2021-05-14T09:37:05.278+05:30", "level":"INFO", "module":"", "message":"info log", "name":"foo", "id":845315}
+{"time":"2021-05-14T09:37:05.279+05:30", "level":"ERROR", "module":"", "message":"error log", "name":"foo", "id":845315}
+{"time":"2021-05-14T09:37:05.285+0g5:30", "level":"ERROR", "module":"", "message":"error log", "stackTrace":[{"callableName":"f3", "moduleName":"log", "fileName":"log.bal", "lineNumber":66}, {"callableName":"f2", "moduleName":"log", "fileName":"log.bal", "lineNumber":62}, {"callableName":"f1", "moduleName":"log", "fileName":"log.bal", "lineNumber":58}, {"callableName":"main", "moduleName":"log", "fileName":"log.bal", "lineNumber":23}], "id":845315, "username":"Alex92"}
```
Comment (daneshk) [inline examples/log-api/log_api.out:21]:
```suggestion
{"time":"2021-05-14T09:37:05.285+05:30", "level":"ERROR", "module":"", "message":"error log", "stackTrace":[{"callableName":"f3", "moduleName":"log", "fileName":"log.bal", "lineNumber":66}, {"callableName":"f2", "moduleName":"log", "fileName":"log.bal", "lineNumber":62}, {"callableName":"f1", "moduleName":"log", "fileName":"log.bal", "lineNumber":58}, {"callableName":"main", "moduleName":"log", "fileName":"log.bal", "lineNumber":23}], "id":845315, "username":"Alex92"}
```

File: examples/log-api/log_api.bal [hunk 15]
```diff
@@ -35,14 +19,34 @@ public function main() {
...
+    // Each module can also be assigned its own log level. To assign a
+    // log level to a module, provide the following entry in the `Config.toml` file:
+    //
+    // ```
+    // [[ballerina.log.modules]]
+    // name = "[ORG_NAME]/[MODULE_NAME]"
+    // level = "[LOG_LEVEL]"
+    // ```
+
+    // By default, log messages are logged to the console in LogFmt format.
```
Comment (daneshk) [inline examples/log-api/log_api.bal:49]: Shall we add these comments to the description file

Comment (MadhukaHarith92) [reply examples/log-api/log_api.bal]: Updated in 54cc3cb

File: examples/log-api/log_api.bal [hunk 16]
```diff
@@ -4,29 +4,13 @@ public function main() {
...
-    // level = "[LOG_LEVEL]"
-    // ```
-
-    // Users can pass any number of key/value pairs which needs to be displayed in the log message.
-    // These can be of `anydata` type including int, string and boolean.
+    // You can pass any number of key/value pairs, which need to be displayed in the log message.
+    // These can be of the `anydata` type including int, string, and boolean.
     log:printInfo("info log", id = 845315, name = "foo", successful = true);
 
     // Users can also pass key/value pairs where the values are function pointers.
```
Comment (MadhukaHarith92) [inline examples/log-api/log_api.bal:16]:
```suggestion
    // You can also pass key/value pairs where the values are function pointers.
```

**Review by daneshk — APPROVED** [review]
LGTM

**Review by praneesha — APPROVED** [review]

**Pull request merged**